generate/generate.go

```diff
@@ -115,6 +115,34 @@ func Load(cfg *config.Tree) ([]LoadResult, error) {
 		res.Files = generated
 		results[i] = res
 	}
+
+	for _, dircfg := range cfg.Root().Dirs() {
+		res := LoadResult{Dir: dircfg.ProjDir()}
+		evalctx, err := eval.NewContext(dircfg.Dir())
+		if err != nil {
+			res.Err = err
+			results = append(results, res)
+			continue
+		}
+
+		var generated []GenFile
+		for _, block := range dircfg.Node.Generate.Files {
+			if block.Context != genfile.RootContext {
+				continue
+			}
+
+			file, err := genfile.Eval(block, evalctx)
+			if err != nil {
+				res.Err = errors.L(res.Err, err).AsError()
+				results = append(results, res)
+				continue
+			}
+
+			generated = append(generated, file)
+			res.Files = generated
+			results = append(results, res)
+		}
```

Comment:
```suggestion
			
		}
		res.Files = generated
		results = append(results, res)
```

Unless we want to have a result with 1 file, then another result with 2 files, etc (for N generate_file blocks).

Reply:
thanks, yeah this is not even tested yet